packages/radio/src/FzRadio.vue

```diff
@@ -0,0 +1,78 @@
+<template>
+  <div class="flex justify-center flex-col w-fit">
+    <input
+      type="radio"
+      :id="label"
+      :value="value || label"
+      :disabled="disabled"
+      :checked="checked"
+      :class="computedClass"
+      :name="name"
+      :required="required"
+      @change="onChange"
+      tabindex="0"
+      ref="radioContainer"
+    />
+    <label :for="label" :class="computedLabelClass">
+      <span class="w-fit">{{ standalone ? "" : label }}</span>
+    </label>
+    <FzRadioErrorText :size="size" v-if="error && $slots.error">
+      <slot name="error" />
+    </FzRadioErrorText>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, onMounted, ref } from "vue";
+import { FzRadioProps } from "./types";
+import { mapSizeToClasses } from "./common";
+import FzRadioErrorText from "./components/FzRadioErrorText.vue";
+
+const props = withDefaults(defineProps<FzRadioProps>(), {
+  size: "md",
+});
+
+const radioContainer = ref<HTMLInputElement | null>(null);
+
+const computedClass = computed(() => ({
+  "radio--small": props.size === "sm",
+  "radio--medium": props.size === "md",
+  "peer h-0 w-0 absolute": true,
+}));
+
+const computedLabelClass = computed(() => [
+  `flex items-start gap-4 h-16 
+  before:content-[''] before:inline-block before:border-solid before:border-1 before:rounded-full before:ml-4
+  peer-checked:before:bg-transparent peer-focus:before:outline peer-checked:before:outline-offset-1 peer-checked:before:outline-1 peer-checked:before:outline-blue-600
+  peer-disabled:text-grey-300
+  peer-disabled:before:border-grey-200 peer-disabled:before:bg-grey-200
+  peer-checked:peer-disabled:before:border-grey-200`,
+  mapSizeToClasses[props.size],
+  props.size === "sm"
```

Comment:
I would follow the same pattern as the other components, splitting "static" and dynamic classes, and using a simpler flow control like switch case for the dynamic part